Controller/Adminhtml/System/Config/Button.php

```diff
@@ -0,0 +1,280 @@
+                'payment/terminal' . $i . '/title',
+                $terminal,
+                $scopeCode,
+                $currentStoreID
+            );
+
+            $this->resourceConfig->saveConfig(
+                'payment/terminal' . $i . '/language',
+                null,
+                $scopeCode,
+                $currentStoreID
+            );
+
+            $this->resourceConfig->saveConfig(
+                'payment/terminal' . $i . '/capture',
+                0,
+                $scopeCode,
+                $currentStoreID
+            );
+
+            $this->resourceConfig->saveConfig(
+                'payment/terminal' . $i . '/terminallogo',
+                '',
+                $scopeCode,
+                $currentStoreID
+            );
+
+            $this->resourceConfig->saveConfig(
+                'payment/terminal' . $i . '/showlogoandtitle',
+                0,
+                $scopeCode,
+                $currentStoreID
+            );
+
+            $this->resourceConfig->saveConfig(
+                'payment/terminal' . $i . '/savecardtoken',
+                0,
+                $scopeCode,
+                $currentStoreID
+            );
+
+            $this->resourceConfig->saveConfig(
+                'payment/terminal' . $i . '/avscontrol',
+                0,
+                $scopeCode,
+                $currentStoreID
+            );
+
+            $this->resourceConfig->saveConfig(
+                'payment/terminal' . $i . '/enforceavs',
+                0,
+                $scopeCode,
+                $currentStoreID
+            );
+
+            $this->resourceConfig->saveConfig(
+                'payment/terminal' . $i . '/avs_acceptance',
+                0,
+                $scopeCode,
+                $currentStoreID
+            );
+
+            $this->resourceConfig->saveConfig(
+                'payment/terminal' . $i . '/sort_order',
+                0,
+                $scopeCode,
+                $currentStoreID
+            );
+
+            $this->resourceConfig->saveConfig(
+                'payment/terminal' . $i . '/terminalname',
+                $terminal,
+                $scopeCode,
+                $currentStoreID
+            );
+
+            $i++;
+        }
+    }
+
+    /**
+     * @param $terminalList array
+     * @param $scopeCode    string
+     *
+     * @return bool
+     */
+    public function checkConfigAlreadyExist($terminalList, $scopeCode, $scopeID)
+    {
+        $i                       = 1;
+        $checkTerminalConfigured = false;
+        foreach ($terminalList as $terminal) {
+            $connection  = $this->_resource->getConnection();
+            $configExist = $connection->fetchAll('SELECT * from core_config_data WHERE path = "payment/terminal' . $i
+                                                 . '/active" AND scope = "' . $scopeCode . '" AND scope_id ='
+                                                 . $scopeID);
```

Comment:
This opens up for a possibility to perform SQL Injection.

Is there a way to use a parameterised request instead?

https://cheatsheetseries.owasp.org/cheatsheets/SQL_Injection_Prevention_Cheat_Sheet.html

Reply:
This has been updated